src/hex.ts

```diff
@@ -55,6 +55,30 @@ export function assertIsStrictHexString(value: unknown): asserts value is Hex {
   );
 }
 
+/**
+ * Validates that the input is a hex address. This utility by default
+ * will return true for hex strings that meet the length requirement
+ * of a hex address, but are not necessarily prefixed with `0x`.
+ *
+ * @param possibleAddress - Input parameter to check against.
+ * @param options - The validation options.
+ * @param options.allowNonPrefixed - If true will allow addresses without `0x` prefix.`
+ * @returns Whether or not the input is a valid hex address.
+ */
+export function isValidHexAddress(
+  possibleAddress: string,
+  { allowNonPrefixed = true } = {},
```

Comment:
Huh. I see that this is how the `controller-utils` version works, but that's too bad. I don't think we ever want non-prefixed hex addresses.

The only place I can think of where we use them would be when calling external libraries or generating signatures where non-prefixed is expected. But even then, we accept them as prefixed, and chop off the `0x` at the last minute.

Reply:
Dropping support for non-prefixed addresses would let us improve the type as well, by using the `Hex` type. That makes it a lot more difficult to pass in the wrong thing.

Reply:
Oh, I see. Do you think we should set only prefixed strings as valid, or are you suggesting changing the default `allowNonPrefixed` value?